pkg/planner/core/logical_plans_test.go

```diff
@@ -1547,32 +1524,632 @@ func TestVisitInfo(t *testing.T) {
 		domain.GetDomain(sctx).MockInfoCacheAndLoadInfoSchema(s.is)
 		builder.ctx.GetSessionVars().SetHashJoinConcurrency(1)
 		_, err = builder.Build(context.TODO(), nodeW)
-		require.NoError(t, err, comment)
+		require.NoError(t, err, tt.sql)
 
-		checkVisitInfo(t, builder.visitInfo, tt.ans, comment)
+		checkVisitInfo(t, builder.visitInfo, tt.ans, tt.sql)
+		require.Nil(t, TryFastPlan(s.ctx, nodeW))
 
 		domain.GetDomain(sctx).StatsHandle().Close()
 	}
 }
 
+func testColumnPrivilegeVisitInfo(t *testing.T) {
+	variable.EnableMDL.Store(false)
+	tests := []struct {
+		sql string
+		ans []visitInfo
+	}{
+		{
+			sql: "insert into t (a) values (1)",
+			ans: []visitInfo{
+				{mysql.InsertPriv, "test", "t", "a", nil, false, nil, false},
+			},
+		},
+		{
+			sql: "delete from t order by a",
+			ans: []visitInfo{
+				{mysql.DeletePriv, "test", "t", "", plannererrors.ErrTableaccessDenied.FastGenByArgs("DELETE", "", "", "t"), false, nil, false},
+				{mysql.SelectPriv, "test", "t", "a", plannererrors.ErrColumnaccessDenied.FastGenByArgs("SELECT", "", "", "a", "t"), false, nil, false},
```

Comment:
This `visitInfo` means that:
> we should have a `SELECT` privilege of column `a` in `test.t`, otherwise an error of `plannererrors.ErrColumnaccessDenied.FastGenByArgs("SELECT", "", "", "a", "t")` will be reported